File: src/dataset/python/rotational-cipher_exercism-ThomasZumsteg/openAI/ChatGPT4_rotational_cipher_v1.py
def rotate(text, key):
    def rot(letter):
        if 'a' <= letter <= 'z':
            return chr((ord(letter) - 97 + key) % 26 + 97)
        if 'A' <= letter <= 'Z':
            return chr((ord(letter) - 65 + key) % 26 + 65)
        return letter
    return ''.join(map(rot, text))

File: src/dataset/javascript/rotational-cipher_exercism-ThomasZumsteg/openAI/ChatGPT4_rotational-cipher_v1.js
class RotationalCipher {
    rotate(text, diff) {
        const lowerA = 'a'.charCodeAt(0), lowerZ = 'z'.charCodeAt(0);
        const upperA = 'A'.charCodeAt(0), upperZ = 'Z'.charCodeAt(0);
        const alphabetSize = 26;

        return text.split('').map(char => {
            const charCode = char.charCodeAt(0);
            if (charCode >= lowerA && charCode <= lowerZ) {
                return String.fromCharCode(((charCode - lowerA + diff) % alphabetSize) + lowerA);
            } else if (charCode >= upperA && charCode <= upperZ) {
                return String.fromCharCode(((charCode - upperA + diff) % alphabetSize) + upperA);
            }
            return char;
        }).join('');
    }
}

module.exports = RotationalCipher;